web/src/hashchange.js

```diff
@@ -303,6 +317,12 @@ function do_hashchange_overlay(old_hash) {
                 // hand-typed a hash.
                 blueslip.warn("missing section for organization");
             }
+            let right_section;
+            if (section === "user-list-admin") {
+                const current_right_section = hash_parser.get_current_nth_hash_section(2);
+                right_section = handle_invalid_users_section_url(current_right_section);
+                settings_panel_menu.org_settings.curr_right_section = right_section;
+            }
             settings_panel_menu.org_settings.activate_section_or_default(section);
```

Comment:
The `right_section` field can be passed here to `activate_section_or_default`. Basically we should just do what we do for `section`.